lib/train/task_repository.dart

```diff
@@ -96,6 +98,65 @@ class Task {
           ),
       };
 
+  Task withSymmetry(Symmetry symmetry) {
+    if (symmetry == Symmetry.identity) {
+      return this;
+    }
+
+    final transformedStonesMap = <wq.Color, ISet<wq.Point>>{};
+    for (final entry in initialStones.entries) {
+      final transformedPoints = entry.value.fold<ISet<wq.Point>>(
+        const ISet<wq.Point>.empty(),
+        (accPoints, point) =>
+            accPoints.add(symmetry.transformPoint(point, boardSize)),
+      );
+      transformedStonesMap[entry.key] = transformedPoints;
+    }
+
+    return Task(
+      id: id,
+      rank: rank,
+      type: type,
+      first: first,
+      boardSize: boardSize,
+      subBoardSize: subBoardSize,
+      topLeft: _transformTopLeft(topLeft, boardSize, subBoardSize, symmetry),
+      initialStones: IMapOfSets(transformedStonesMap),
+      variationTree: variationTree.withSymmetry(symmetry, boardSize),
+    );
+  }
+
+  wq.Point _transformTopLeft(
+      topLeft, int boardSize, int subBoardSize, Symmetry symmetry) {
+    if (symmetry == Symmetry.identity) return topLeft;
+
+    final (x1, y1) = topLeft;
+    final (x2, y2) =
+        (topLeft.$1 + subBoardSize - 1, topLeft.$2 + subBoardSize - 1);
+
+    final wq.Point topRight = (x2, y1);
+    final wq.Point bottomRight = (x2, y2);
+    final wq.Point bottomLeft = (x1, y2);
+
+    final tlS = symmetry.transformPoint(topLeft, boardSize);
+    final trS = symmetry.transformPoint(topRight, boardSize);
+    final blS = symmetry.transformPoint(bottomRight, boardSize);
+    final brS = symmetry.transformPoint(bottomLeft, boardSize);
+
+    var topLeftS = (
+      min(tlS.$1, min(trS.$1, min(blS.$1, brS.$1))),
+      min(tlS.$2, min(trS.$2, min(blS.$2, brS.$2)))
+    );
+
+    topLeftS = (max(0, topLeftS.$1), max(0, topLeftS.$2));
```

Comment:
this clamping seems unnecessary (in theory). Let's remove it and directly return the value of `topLeftS`.